FormalConjectures/ErdosProblems/503.lean

```diff
@@ -0,0 +1,86 @@
+/-
+Copyright 2025 The Formal Conjectures Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-/
+
+import FormalConjectures.Util.ProblemImports
+
+/-!
+# Erdős Problem 503
+
+*Reference:* [erdosproblems.com/503](https://www.erdosproblems.com/503)
+-/
+
+namespace Erdos503
+
+abbrev ESpace (n : ℕ) := EuclideanSpace ℝ (Fin n)
+
+def isIsosceles {n : ℕ} (x y z : ESpace n) :=
+  dist x y = dist y z ∨ dist y z = dist x z ∨ dist x y = dist x z
+
+def isoscelesSet {n : ℕ} (A : Set (ESpace n)) := A.Triplewise (λ x y z ↦ isIsosceles x y z)
+
+/--
+What is the size of the largest $A \subseteq \mathbb{R}^n$ such that every three points from $A$
+determine an isosceles triangle? That is, for any three points $x$, $y$, $z$ from $A$, at least two
+of the distances $|x - y|$, $|y - z|$, $|x - z|$ are equal.
+-/
+@[category research open, AMS 51]
+theorem erdos_503 {n : ℕ} : ∃ A : Set (ESpace n), isoscelesSet A ∧ (∀ B : Set (ESpace n),
+    isoscelesSet B → B.ncard ≤ A.ncard) ∧ A.ncard = answer(sorry) := by
```

Comment:
one could use `IsGreatest` here
also below. 
But perhaps even better define the largest such that separately using `iSup` or `sSup` then we can use it all over the place?